tests/functional/aws-node-sdk/test/bucket/putBucketLifecycle.js

```diff
@@ -505,7 +508,11 @@ describe('aws-sdk test put bucket lifecycle', () => {
                 }];
                 const params = getParams(transitions);
                 s3.putBucketLifecycleConfiguration(params, err => {
-                    assert.ifError(err);
+                    if (isSupported) {
+                        assert.ifError(err);
+                    } else {
+                        assert.strictEqual(err && err.code, 'NotImplemented');
```

Comment:
Would be better to have separate tests, expecting an error when the test is called "is supported" can lead to strange things...

Reply:
I've changed the name of the test dynamically

Reply:
not sure this is the most readable, esp. since a single test (or suite!) checking `putBucketLifecycleConfiguration` returns NotImplemented is probably enough when transitions are not supported, without complexifying each test and label...